src/iiiftilesource.js

```diff
@@ -55,6 +55,16 @@ $.IIIFTileSource = function( options ){
 
     options.tileSizePerScaleFactor = {};
 
+    this.usedFormat = "jpg";
+    if ( this.formatHints ) {
```

Comment:
Does this require a `formatHints` property on the incoming JSON? If so, this is not specified by IIIF, and thus I think there should be an API to set it on the tileSource directly.

Reply:
it requires it to be activated yes. We implemented that in our image services so that all viewers pass it to OSD. I can implement a function to set it, I'm just unsure where it should appear... something at the level of [this one](https://github.com/openseadragon/openseadragon/blob/master/src/iiiftilesource.js#L194)?

Reply:
I'm :-1: on code in a IIIF tilesource that relies on custom extensions without an API equivalent.

Could it be passed in the options constructor param?
https://github.com/openseadragon/openseadragon/blob/master/src/tilesource.js#L55

Reply:
Sure, I'm happy to do that but I don't really understand what it takes to do that in terms of code...? is it just documenting 

```
 * @param {Object} [options.formatHints]
 *      An array of extensions to use for image URLs, by descending order of preference. The extensions are tested against the supported formats, in case none of the extension values is supported, the default (jpg) will be used.
```

around the lines you pointed to or are there other steps to take?

Reply:
I added something, tell me if it works

Reply:
@azaroth42 is that ok now?

Reply:
I'm afraid I don't have the time to validate, but it looks correct. Can you write a test to verify that it works as expected?